src/components/AutoComplete.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, createElement } from "react";
+
+import * as Autosuggest from "react-autosuggest";
+
+export interface Suggestion {
+    name: string;
+    newValue: string;
+    suggestionValue: string;
+    value: string;
+}
+
+export interface AutoCompleteProps {
+    addTag: (tag: string) => void;
+    inputPlaceholder: string;
+    lazyLoad?: boolean;
+    fetchSuggestions?: () => void;
+    suggestionValue?: string;
+    suggestions: Suggestion[];
+}
+
+interface AutoCompleteState {
+    lazyLoaded: boolean;
+    suggestions: Suggestion[];
+    value: string;
+}
+
+export class AutoComplete extends Component<AutoCompleteProps, AutoCompleteState> {
+
+    constructor(props: AutoCompleteProps) {
+        super(props);
+
+        this.state = {
+            lazyLoaded: false,
+            suggestions: [],
+            value: ""
+        };
+        this.onChange = this.onChange.bind(this);
+        this.getSuggestions = this.getSuggestions.bind(this);
+        this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this);
+        this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this);
+        this.getSuggestionValue = this.getSuggestionValue.bind(this);
+        this.getSuggestions = this.getSuggestions.bind(this);
+        this.renderSuggestion = this.renderSuggestion.bind(this);
+        this.onSuggestionSelected = this.onSuggestionSelected.bind(this);
+    }
+
+    render() {
+        const { value, suggestions } = this.state;
+
+        const inputProps = {
+            onChange: this.onChange,
+            placeholder: this.props.inputPlaceholder,
+            value
+        };
+
+        if (!this.state.lazyLoaded && this.props.lazyLoad) {
+            return createElement("span", {
+                className: "glyphicon glyphicon-refresh glyphicon-refresh-animate",
+                id: "tag-loader"
+            });
+        } else {
+
+            return createElement(Autosuggest, {
+                getSuggestionValue: this.getSuggestionValue,
+                inputProps,
+                onSuggestionSelected: this.onSuggestionSelected,
+                onSuggestionsClearRequested: this.onSuggestionsClearRequested,
+                onSuggestionsFetchRequested: this.onSuggestionsFetchRequested,
+                renderSuggestion: this.renderSuggestion,
+                suggestions
+            });
+        }
+    }
+
+    componentDidMount() {
+        setTimeout(() => {
+            this.setUpSuggestions(this.props);
+            this.setState({ lazyLoaded: true
+            });
```

Comment:
one line `this.setState({ lazyLoaded: true });`

Reply:
Done